core/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -435,15 +440,16 @@ public CircuitBreakerService breakerService() {
 
     private <T extends AbstractBigArray> T resizeInPlace(T array, long newSize) {
         final long oldMemSize = array.ramBytesUsed();
+        final long estimatedIncreaseInBytes = array.ramBytesEstimated(newSize) - oldMemSize;
+        adjustBreaker(estimatedIncreaseInBytes, false);
         array.resize(newSize);
-        adjustBreaker(array.ramBytesUsed() - oldMemSize);
```

Comment:
could we add an assert that array.ramBytesUsed() is equal to the estimate we computed previously?